crates/ruff_db/src/diagnostic/render.rs

```diff
@@ -67,62 +68,130 @@ impl std::fmt::Display for DisplayDiagnostic<'_> {
-            )?;
+        match self.config.format {
+            DiagnosticFormat::Concise => {
+                let (severity, severity_style) = match self.diag.severity() {
+                    Severity::Info => ("info", stylesheet.info),
+                    Severity::Warning => ("warning", stylesheet.warning),
+                    Severity::Error => ("error", stylesheet.error),
+                    Severity::Fatal => ("fatal", stylesheet.error),
+                };
 
-            if let Some(span) = self.diag.primary_span() {
                 write!(
                     f,
-                    " {path}",
-                    path = fmt_styled(span.file().path(self.resolver), stylesheet.emphasis)
+                    "{severity}[{id}]",
+                    severity = fmt_styled(severity, severity_style),
+                    id = fmt_styled(self.diag.id(), stylesheet.emphasis)
                 )?;
-                if let Some(range) = span.range() {
-                    let diagnostic_source = span.file().diagnostic_source(self.resolver);
-                    let start = diagnostic_source
-                        .as_source_code()
-                        .line_column(range.start());
 
+                if let Some(span) = self.diag.primary_span() {
                     write!(
                         f,
-                        ":{line}:{col}",
-                        line = fmt_styled(start.line, stylesheet.emphasis),
-                        col = fmt_styled(start.column, stylesheet.emphasis),
+                        " {path}",
+                        path = fmt_styled(span.file().path(self.resolver), stylesheet.emphasis)
                     )?;
+                    if let Some(range) = span.range() {
+                        let diagnostic_source = span.file().diagnostic_source(self.resolver);
+                        let start = diagnostic_source
+                            .as_source_code()
+                            .line_column(range.start());
+
+                        write!(
+                            f,
+                            ":{line}:{col}",
+                            line = fmt_styled(start.line, stylesheet.emphasis),
+                            col = fmt_styled(start.column, stylesheet.emphasis),
+                        )?;
+                    }
+                    write!(f, ":")?;
+                }
+                writeln!(f, " {message}", message = self.diag.concise_message())
+            }
+            DiagnosticFormat::Json => {
+                // Placeholder for JSON serialization logic
+                // This will be replaced with actual serialization code
+                let json_diagnostic = JsonDiagnostic::from_diagnostic(self.diag, self.resolver);
+                match serde_json::to_string(&json_diagnostic) {
+                    Ok(json_string) => write!(f, "{}", json_string),
+                    Err(_) => Err(std::fmt::Error), // Or handle error more gracefully
```

Comment:
Nit: I think this should be `let json = serde_json::to_string(&json_diagnostic).map_err(|_| std::fmt::Error)?;` and then `write!(f, "{json}")`.